### PR TITLE
frontend: created and used lazy loaded qrcode scanner hook

### PR DESCRIPTION
**Pull request by shonsirsha**
To make lazy loaded QR code scanner more reusable, we can create a hook for it.

### CODE REVIEW

**Comment by shonsirsha** [comment]
This (at least) can be re-used for wallet connect qr code scanning. 🙂

**Review by thisconnect — CHANGES_REQUESTED** [review]
nice test !

File: frontends/web/src/hooks/use-qr-code-reader.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  useEffect(() => {
+    import('../components/qrcode/qrreader')
+      .then(({ BrowserQRCodeReader }) => {
+        if (!qrCodeReaderRef.current) {
+          qrCodeReaderRef.current = new BrowserQRCodeReader();
+        }
+
+        qrCodeReaderRef.current.getVideoInputDevices()
+          .then(videoInputDevices => {
+            setHasCamera(videoInputDevices.length > 0);
```
[the remark this answers is not in the document]
Comment (shonsirsha) [reply frontends/web/src/hooks/use-qr-code-reader.ts]: good point - always nice to be safe. Changes made & rebased. PTAL 🙏

**Comment by shonsirsha** [comment]
Renamed the hook and test file to `qrcodescanner` & `qrcodescanner.test` respectively to keep consistency between hook files naming.

**Review by thisconnect — APPROVED** [review]
LGTM 👍

**Pull request merged**